src/vs/workbench/contrib/scm/browser/scmViewService.ts

```diff
@@ -184,6 +186,14 @@ export class SCMViewService implements ISCMViewService {
 		}
 
 		storageService.onWillSaveState(this.onWillSaveState, this, this.disposables);
+
+		// Maintain repository selection when the extension host restarts.
+		// Extension host is restarted after installing an extension update
+		// or during a profile switch.
+		extensionService.onWillStop(() => {
+			this.onWillSaveState();
```

Comment:
Confusing naming here that I avoid. I suggest something like `handleOnWillSaveState`